pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -642,6 +642,42 @@ func BuildKubernetesMasterConfig(
 	return kmaster, nil
 }
 
+// BuildKubernetesControllersConfig builds a Kubernetes master config only for
+// controllers.
+func BuildKubernetesControllersConfig(
+	masterConfig configapi.MasterConfig,
+	kubeClient kclientset.Interface,
+	internalKubeClient kinternalclientset.Interface,
+	externalKubeInformers kexternalinformers.SharedInformerFactory,
+) (*MasterConfig, error) {
+	if masterConfig.KubernetesMasterConfig == nil {
+		return nil, errors.New("insufficient information to build KubernetesMasterConfig")
+	}
+
+	controllerServer, cloud, err := buildControllerManagerServer(masterConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	schedulerServer, err := buildSchedulerServer(masterConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kmaster := &MasterConfig{
```

Comment:
I'd rather see a new struct built that contains what we need and build that.  Slimming down is never done and side-effects get weird.